File: challenges/candies/java/Candies.java
import java.util.stream.IntStream;

public class Candies {
	public static void main(String[] args) {
		int[] input0 = new int[] {3, 1, 10, 4, 4, 9, 8, 3, 7, 3};
		int[] input1 = new int[] {10, 2, 5, 4, 15, 100, 2, 3, 10, 23};
		int[] expectedOutput = new int[] {9, 2, 0, 4, 12, 99, 0, 3, 7, 21};
		assert input0.length == expectedOutput.length : String.format("# input0 = %d, # expectedOutput = %d", input0.length, expectedOutput.length);
		assert input1.length == expectedOutput.length : String.format("# input1 = %d, # expectedOutput = %d", input1.length, expectedOutput.length);
		IntStream.range(0, expectedOutput.length).forEach(i -> {
			int actualOutput = candies(input0[i], input1[i]);
			assert actualOutput == expectedOutput[i] : String.format("candies(%d, %d) returned %d, but expected %d", input0[i], input1[i], actualOutput, expectedOutput[i]);
		});
		System.out.println(String.format("PASSES %d out of %d tests", expectedOutput.length, expectedOutput.length));
	}

	public static int candies(int n, int m) {
		return m / n * n;
	}
}

File: challenges/candies/octave/candies.m
function r = candies(n, m)
	r = floor(m/n)*n;
endfunction
%!assert(candies(3, 10), 9)
%!assert(candies(1, 2), 2)
%!assert(candies(10, 5), 0)
%!assert(candies(4, 4), 4)
%!assert(candies(4, 15), 12)
%!assert(candies(9, 100), 99)
%!assert(candies(8, 2), 0)
%!assert(candies(3, 3), 3)
%!assert(candies(7, 10), 7)
%!assert(candies(3, 23), 21)